icicle/utils/vec_ops.cu

```diff
@@ -146,4 +185,20 @@ namespace vec_ops {
     return Sub<curve_config::scalar_t>(vec_a, vec_b, n, config, result);
   }
 
+  /**
+   * Extern version of transpose_batch function with the template parameter
+   * `E` being the [scalar field](@ref scalar_t) of the curve given by `-DCURVE` env variable during build.
+   * @return `cudaSuccess` if the execution was successful and an error code otherwise.
+   */
+  extern "C" cudaError_t CONCAT_EXPAND(CURVE, TransposeBatch)(
+    curve_config::scalar_t* input,
```

Comment:
better make the input const pointer (const scalar*). This is also true for the kernels.
In some cases (for example polynomials) the internal memory can only be accessed by const so this way means this API cannot be used.